magento/pwa-studio
# Fixes Cart page sign out error when cart does not exist in cache

## Description

This PR provides a fallback empty object (`{}`) to avoid a destructuring exception thrown when the cart does not exist in our Apollo cache.

This surfaced on the `/cart` page after the user signed out: the cache was cleared out using Apollo client's `resetStore`. But `resetStore` also causes all active queries to be refetched. `GiftOptions` has one such query that was querying for the cart in the cache. Because it was just cleared out via `resetStore`, the query got back `undefined`, which the code then attempted to destructure.

Providing the fallback empty object in the case of `undefined` returns the expected result (`undefined` values for `include_gift_receipt`, `include_printed_card`, and `gift_message`), and results in the ability to sign out from the cart page without error.

## Related Issue





Closes PWA-368.

## Acceptance 



### Verification Stakeholders


@revanth0212 

### Specification



### Verification Steps


1. Sign in
1. Ensure an item is added to your cart
1. Navigate to the `/cart` page
1. Verify that you can sign out successfully without error
1. Sign in
1. Verify that your cart is restored to its previous state

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
